cli/cmd/release_create.go

```diff
@@ -166,3 +163,42 @@ func (r *runners) releaseCreate(cmd *cobra.Command, args []string) error {
 
 	return nil
 }
+
+// todo:dex
+// this has some kots-only logic that is leaking through the Client interface
+// abstraction. Need to clean this up, probably a GetOrCreateChannel method on
+// all the Client impls
+func (r *runners) getOrCreateChannelForPromotion() (string, error) {
+	channels, err := r.api.ListChannels(r.appID, r.appType)
+	if err != nil {
+		return "", err
+	}
+
+	promoteChannelIDs := make([]string, 0)
+	for _, c := range channels {
+		if c.ID == r.args.createReleasePromote || c.Name == r.args.createReleasePromote {
+			promoteChannelIDs = append(promoteChannelIDs, c.ID)
+		}
+	}
+
+	if len(promoteChannelIDs) == 0 {
+
+		// get-or-create only supported for KOTS apps at the moment
+		shouldCreateChannel := r.args.createReleasePromoteEnsureChannel && r.appType == "kots"
+
+		if shouldCreateChannel {
+			channelID, err := r.kotsAPI.CreateChannel(r.appID, r.args.createReleasePromote, "")
```

Comment:
is this arg createReleasePromote or createReleasePromoteDir for Kots app?

Reply:
`createReleasePromote` is the *channel name* to promote the release to after creating -- I'd be open to updating the variable name in a followup PR if you think there's a way to make that more clear